### ISSUE TITLE
Fetching recommended shards is broken

### ISSUE THREAD

**Issue by appellation** [hydrabolt/discord.js#1526]
**Please describe the problem you are having in as much detail as possible:**
I get an error `Cannot read property 'bot' of undefined` when using the sharding manager.  The error comes from `Util.fetchRecommendedShards` is the result of `Constants.Endpoints.gateway` being undefined.

**Include a reproducible code sample here, if possible:**
```js
const manager = new ShardingManager(file, { token: 'token' });
manager.spawn();
```

**Further details:**

- discord.js version: 12.0.0
- node.js version: 7.8.0
- Operating system: Windows
- Priority this issue should have – please be realistic and elaborate if possible: sharding is broken, so probably high

- [x] I have also tested the issue on latest master, commit hash: 4292134647608206e2b3b379485a5494854c648e

**Comment by JMTK** [comment]
FYI short term fix below(you don't have to use unirest, just need to use a request library:
```js
unirest
    .get('https://discordapp.com/api/gateway/bot')
    .header('Authorization', "Bot " + Config.Auth.Token)
    .header('Accept', 'application/json')
    .end(function(response) {
        console.log("MASTER>", "Shard count is", response.body.shards);
        sharder.spawn(Number(response.body.shards));
    }
);
```

**Comment by devsnek** [comment]
use snekfetch <.<

**Comment by PhoenixShay** [comment]
https://github.com/hydrabolt/discord.js/pull/1532

**Comment by iCrawl** [comment]
Fixed in 22e8237b
